monix-eval/shared/src/main/scala/monix/eval/Coeval.scala

```diff
@@ -394,6 +394,20 @@ object Coeval {
   def fromTry[A](a: Try[A]): Coeval[A] =
     Attempt.fromTry(a)
 
+  /** Aquires a `Closeable` and closes it after the `Coeval` has been run.
+    */
+  def bracket[R <: java.io.Closeable](aquire: => R): Coeval[R] = defer {
+    val resource: R = aquire
+    now(resource).doOnFinish(_ => delay(resource.close))
```

Comment:
Well, this isn't going to work, as `Coeval` is a simpler abstraction than the `Process` in FS2. `doOnFinish` is equivalent to this:

```scala
def doOnFinish[A](source: Coeval[A])(f: Option[Throwable] => Coeval[Unit]): Coeval[A] =
  source.materialize.flatMap {
    case Now(value) => f(None)
    case Error(ex) => f(Some(ex))
  }
```

In other words, this isn't end-of-the-world evaluation. For `Task` and `Coeval` you can't specify that in advance. What you can do instead is something like this:

```scala
def using[R <: java.io.Closable, A](resource: => R)(f: R => A): Coeval[A] =
  Coeval.defer {
    val ref = resource
    val close = Task.eval(ref.close())
    Coeval.eval(f(ref)).doOnFinish(_ => close)
  }
```